modelling/src/main/java/org/axonframework/modelling/command/ModelLoader.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.modelling.command;
+
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Loader of a model of type {@code M} with identifier of type {@code ID}.
+ *
+ * @param <I> The type of the identifier of the model
+ * @param <M>  The type of model to load
+ */
+@FunctionalInterface
+public interface ModelLoader<I, M> {
```

Comment:
When I was doing the review it was not obvious for me what is the purpose of ModeLoader#load and how it differs from ModelContainer#resolve. Maybe put the ModelLoader inside `ModelRegistry`? 
We can even rename it to `ModelSupplier`? Because we define "how to supply the model" while definiting it in the ModelRegistry.

Reply:
Yeah I had it in the interface at first. Renaming it to `ModelSupplier` and putting it inside the interface :)